### PR TITLE
Fix `?until` in BBS commit contexts

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

In Bitbucket server, the commits view works a bit differently than on https://bitbucket.org. Instead of containing the concerned branch name in the URL path, it is stored in the `until` search parameter. We did not account for that. 

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 72bc2b0</samp>

This pull request enhances the `BitbucketServerContextParser` class and its tests to support commit context URLs with branch names. This allows users to open Gitpod workspaces from such URLs and see the correct branch and commit information.

</details>

## Related Issue(s)

Fixes EXP-844

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-bbs-until-fix</li>
	<li><b>🔗 URL</b> - <a href="https://ft-bbs-until-fix.preview.gitpod-dev.com/workspaces" target="_blank">ft-bbs-until-fix.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-bbs-until-fix-gha.19096</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-bbs-until-fix%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## How to test

I had to do some code gymnastics to get the tests working properly with our BBS instance, but here they run, successfully.

image

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by AlexTugarev** [comment]
Looking into this now...

File: components/server/src/bitbucket-server/bitbucket-server-context-parser.spec.ts [hunk 1]
```diff
@@ -186,6 +186,33 @@ class TestBitbucketServerContextParser {
         });
     }
 
+    @test async test_commit_context_02() {
```
Comment (AlexTugarev) [inline components/server/src/bitbucket-server/bitbucket-server-context-parser.spec.ts:189]: nit: not necessary to change, just find it hard to navigate through the different test cases. here we're talking about the Commits page of BBS, not the page to show the single commit.

**Comment by AlexTugarev** [comment]
It's be best to test that first. 

Code wise it looks fine. 

To recap: BBS has the `?at` param on the Browse page, and the `?until` on the Commits page. 

| Browse | Commits |
|--------|--------|
| Screenshot 2023-10-31 at 11 25 57 | Screenshot 2023-10-31 at 11 26 12 |

File: components/server/src/bitbucket-server/bitbucket-server-context-parser.spec.ts (hunk 1 above)
Comment (AlexTugarev) [reply components/server/src/bitbucket-server/bitbucket-server-context-parser.spec.ts]: No need to change nits right away, it's interrupting testing with the preview env :-(

Comment (filiptronicek) [reply components/server/src/bitbucket-server/bitbucket-server-context-parser.spec.ts]: Agree the test name is confusing, changed it to "branch_context"

**Review by AlexTugarev — APPROVED** [review]
Successfully tested with https://ft-bbs-until-fix.preview.gitpod-dev.com/#https://bitbucket.gitpod-dev.com/users/svenefftinge/repos/browser-extension-test/commits?until=refs%2Fheads%2FKAN-1-my-issue&merges=include

Screenshot 2023-10-31 at 12 37 38

**Pull request merged**